leasing/management/commands/import_level_check_index.py

```diff
@@ -0,0 +1,299 @@
+        "url": "https://pxdata.stat.fi:443/PxWeb/api/v1/en/StatFin/ashi/statfin_ashi_pxt_13mp.px",
+        "code": "ketj_P_QA_T",
+    },
+]
+
+
+class ColumnItem(TypedDict, total=False):
+    """Details of a column in API response data.
+
+    Based on section 4.5 in API docs https://pxdata.stat.fi/API-description_SCB.pdf
+
+    Keys:
+        code: Identifier for the column.
+        text: Textual display name for the column.
+        type: Type of the column: "d" = dimension, "t" = time, "c" = measure. Optional. Default: "d".
+        unit: Unit for the measures. Only for type "c". Optional.
+        comment: Optional.
+    """
+
+    code: str
+    text: str
+    type: str
+    unit: str
+    comment: str
+
+
+class CommentItem(TypedDict):
+    """A single comment item for a value in API response data.
+
+    Keys:
+        variable: Identifies the column.
+        value: Identifies the value.
+        comment: A comment for the specified value.
+    """
+
+    variable: str
+    value: str
+    comment: str
+
+
+class DataPoint(TypedDict):
+    """A single data point from the API response data.
+
+    All keys and values should be in same order as "columns" items in API response data.
+
+    Keys:
+        key: List of all dimension and time column values.
+        values: List of all measure column values.
+    """
+
+    key: list[str]
+    values: list[str]
+
+
+class MetadataItem(TypedDict):
+    """Metadata details for the database table that the response is based on."""
+
+    updated: str
+    label: str
+    source: str
+
+
+class ResponseData(TypedDict):
+    """The entire data part of the API response."""
+
+    columns: list[ColumnItem]
+    comments: list[CommentItem]
+    data: list[DataPoint]
+    metadata: list[MetadataItem]
+
+
+class ResponseDataError(Exception):
+    """When the data from API response might cause the program to malfunction."""
+
+    pass
+
+
+class Command(BaseCommand):
+    help = "Import indexes related to Level Check (Tasotarkistus) from Tilastokeskus database."
+
+    def handle(self, *args, **options):
+        for index_input in INDEXES_TO_IMPORT:
+            self.stdout.write(f'Index: "{index_input["name"]}"')
+
+            index_data = _get_index_data(index_input["url"], index_input["code"])
+            columns = index_data.get("columns", [])
+
+            # Find the order of the necessary columns to avoid future failures
+            # if columns are added, removed, or moved.
+            try:
+                quarter_key_pos = _find_key_position(columns, "Vuosineljännes")
+                region_key_pos = _find_key_position(columns, "Alue")
+                index_column_pos = _find_column_position(columns, index_input["code"])
+                number_value_pos = _find_value_position(columns, index_input["code"])
+            except ResponseDataError as e:
+                self.stderr.write(f"Error: {e}. Skipping this index import")
```

Comment:
For the output, would be preferable to replace `this` with the name of the index, e.g. the code

Reply:
I assume this helps with Sentry and other error reporting tools? In that case, good idea. I did not specify the index here originally because the index name is written to `stdout` earlier in the same block.

Reply:
Fixed.